server/src/test/java/org/opensearch/search/query/QueryProfilePhaseTests.java

```diff
@@ -615,12 +615,22 @@ public void testTerminateAfterEarlyTermination() throws Exception {
                 assertThat(query.getTimeBreakdown().get("score"), greaterThan(0L));
                 assertThat(query.getTimeBreakdown().get("score_count"), greaterThan(0L));
                 if (executor != null) {
-                    assertThat(query.getTimeBreakdown().get("max_score"), greaterThan(0L));
-                    assertThat(query.getTimeBreakdown().get("min_score"), greaterThan(0L));
-                    assertThat(query.getTimeBreakdown().get("avg_score"), greaterThan(0L));
-                    assertThat(query.getTimeBreakdown().get("max_score_count"), greaterThan(0L));
-                    assertThat(query.getTimeBreakdown().get("min_score_count"), greaterThan(0L));
-                    assertThat(query.getTimeBreakdown().get("avg_score_count"), greaterThan(0L));
+                    long maxScore = query.getTimeBreakdown().get("max_score");
+                    long minScore = query.getTimeBreakdown().get("min_score");
+                    long avgScore = query.getTimeBreakdown().get("avg_score");
+                    long maxScoreCount = query.getTimeBreakdown().get("max_score_count");
+                    long minScoreCount = query.getTimeBreakdown().get("min_score_count");
+                    long avgScoreCount = query.getTimeBreakdown().get("avg_score_count");
+                    assertThat(maxScore, greaterThan(0L));
+                    assertThat(minScore, greaterThanOrEqualTo(0L));
```

Comment:
can you share an example with and without concurrent search enabled, where the scores are coming out to be 0 for some subquery ?

Reply:
For example, consider the `BooleanQuery` "foo:bar foo:baz" with the seed id `E6C322CFA1FDE270:89327DD80FA3AC95`. The response has only `1` total hit and the score for the second child query (i.e. "foo:bar") is zero.
``` json
{
  "took": ...,
  "timed_out": false,
  "_shards": {...},
  "hits": {...},
  "profile": {
    "shards": [
      {
        "id": ...,
        "inbound_network_time_in_millis": 0,
        "outbound_network_time_in_millis": 0,
        "searches": [
          {
            "query": [
              {
                "type": "BooleanQuery",
                "description": "foo:bar foo:baz",
                "time_in_nanos": 445667,
                "breakdown": {
                  "set_min_competitive_score_count": 0,
                  "match_count": 0,
                  "shallow_advance_count": 0,
                  "set_min_competitive_score": 0,
                  "next_doc": 1870,
                  "match": 0,
                  "next_doc_count": 2,
                  "score_count": 7,
                  "compute_max_score_count": 0,
                  "compute_max_score": 0,
                  "advance": 1787,
                  "advance_count": 1,
                  "score": 10874,
                  "build_scorer_count": 2,
                  "create_weight": 15831,
                  "shallow_advance": 0,
                  "create_weight_count": 1,
                  "build_scorer": 62757
                },
                "children": [
                  {
                    "type": "TermQuery",
                    "description": "foo:bar",
                    "time_in_nanos": 221542,
                    "breakdown": {
                      "set_min_competitive_score_count": 0,
                      "match_count": 0,
                      "shallow_advance_count": 0,
                      "set_min_competitive_score": 0,
                      "next_doc": 597,
                      "match": 0,
                      "next_doc_count": 1,
                      "score_count": 7,
                      "compute_max_score_count": 0,
                      "compute_max_score": 0,
                      "advance": 819,
                      "advance_count": 1,
                      "score": 6958,
                      "build_scorer_count": 3,
                      "create_weight": 4999,
                      "shallow_advance": 0,
                      "create_weight_count": 1,
                      "build_scorer": 9580
                    }
                  },
                  {
                    "type": "TermQuery",
                    "description": "foo:baz",
                    "time_in_nanos": 168125,
                    "breakdown": {
                      "set_min_competitive_score_count": 0,
                      "match_count": 0,
                      "shallow_advance_count": 0,
                      "set_min_competitive_score": 0,
                      "next_doc": 114,
                      "match": 0,
                      "next_doc_count": 1,
                      "score_count": 0,
                      "compute_max_score_count": 0,
                      "compute_max_score": 0,
                      "advance": 158,
                      "advance_count": 1,
                      "score": 0,
                      "build_scorer_count": 3,
                      "create_weight": 959,
                      "shallow_advance": 0,
                      "create_weight_count": 1,
                      "build_scorer": 2976
                    }
                  }
                ]
              }
            ],
            "rewrite_time": ...,
            "collector": [...]
          }
        ],
        "aggregations": []
      }
    ]
  }
}
```

Reply:
Thanks